server/src-lib/Hasura/RQL/DML/Select/Types.hs

```diff
@@ -76,9 +76,11 @@ data ArrSel
   deriving (Show, Eq)
 
 type ArrSelFlds = Fields ArrSel
+type AnnArgs = Fields PGColValue
 
 data AnnFld
   = FCol !PGColInfo
+  | FColArg !PGColInfo !AnnArgs
```

Comment:
Let's merge `FCol` and `FColArg` and make it more generic. Something like `FCol !(Maybe ColOp)`. This conveys, we're selecting a `column` that `may` required an `operation` to be applied. 
```haskell
data ColOp
  = ColOp
  { _coOp :: !S.SQLOp -- operation to be applied
  , _coExp :: !S.SQLExp -- RHS expression
  }
```

Reply:
Ok. I create new constructor because I am afraid of breaking changes with old code base.